snuba/clickhouse/astquery.py

```diff
@@ -19,7 +19,7 @@ class AstClickhouseQuery(ClickhouseQuery):
     AST.
     """
 
-    def __init__(self, query: Query, settings: RequestSettings,) -> None:
+    def __init__(self, query: PhysicalQuery, settings: RequestSettings,) -> None:
```

Comment:
I called it PhysicalQuery since we are talking about physical model and logical model. Really no strong opinion on this. If someone prefers StorageQuery I will change it.

Reply:
We have some references to StorageQuery in other places (e.g. StorageQuerySplitStrategy, StorageQueryPlanBuilder) so it might be a bit more consistent.

Reply:
No strong preference here. I agree that consistency is good regardless of which 

 A disadvantage of `StorageQuery` is the implication that the query is run _on_ the storage rather than on the `Cluster` in coordination with the storage(s). Also "storage" being singular might send the wrong message as well since a query can reference multiple storages.

I think in an ideal scenario we would split the logical and physical domains in code and have `snuba.logical.Query` and `snuba.physical.Query`. The only thing that bridges between them would be the translation of `logical.Query` to `physical.Query` and everything else would be unambiguous by its location in the code: `snuba.logical.*` would reference the `logical.Query` exclusively, and similar for the physical query respectively. I am not sure things are clear cut enough yet to justify that scope of a change yet but that's the direction I'd like to see.

Reply:
makes sense. So that means Physical over Storage (whether it is in the class name or in the module name).
@lynnagara I think we can easily rename StorageQueryPlanBuilder. That came before we started talking about logical and physical.

Reply:
This should be addressed now:
- physical = clickhouse
- we have a logical query and a clickhouse query (same Query name in different packages)
- we have a ClickhouseQueryPlan and ClickhouseQueryPlanBuilder